aci/resource_aci_dhcprelayp.go

```diff
@@ -0,0 +1,343 @@
+
+	return resourceAciDHCPRelayPolicyRead(d, m)
+}
+
+func resourceAciDHCPRelayPolicyUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] DHCPRelayPolicy: Beginning Update")
+
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+
+	TenantDn := d.Get("tenant_dn").(string)
+
+	dhcpRelayPAttr := models.DHCPRelayPolicyAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		dhcpRelayPAttr.Annotation = Annotation.(string)
+	} else {
+		dhcpRelayPAttr.Annotation = "{}"
+	}
+	if Mode, ok := d.GetOk("mode"); ok {
+		dhcpRelayPAttr.Mode = Mode.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		dhcpRelayPAttr.NameAlias = NameAlias.(string)
+	}
+	if Owner, ok := d.GetOk("owner"); ok {
+		dhcpRelayPAttr.Owner = Owner.(string)
+	}
+	dhcpRelayP := models.NewDHCPRelayPolicy(fmt.Sprintf("relayp-%s", name), TenantDn, desc, dhcpRelayPAttr)
+
+	dhcpRelayP.Status = "modified"
+
+	err := aciClient.Save(dhcpRelayP)
+
+	if err != nil {
+		return err
+	}
+	d.Partial(true)
+
+	d.SetPartial("name")
+
+	d.Partial(false)
+
+	checkDns := make([]string, 0, 1)
+
+	if d.HasChange("relation_dhcp_rs_prov") {
+		oldRel, newRel := d.GetChange("relation_dhcp_rs_prov")
+		oldRelSet := oldRel.(*schema.Set)
+		newRelSet := newRel.(*schema.Set)
+		relToCreate := toStringList(newRelSet.Difference(oldRelSet).List())
+
+		for _, relDn := range relToCreate {
+			checkDns = append(checkDns, relDn)
+		}
+	}
+
+	d.Partial(true)
+	err = checkTDn(aciClient, checkDns)
+	if err != nil {
+		return err
+	}
+	d.Partial(false)
+
+	if d.HasChange("relation_dhcp_rs_prov") {
+		oldRel, newRel := d.GetChange("relation_dhcp_rs_prov")
+		oldRelSet := oldRel.(*schema.Set)
+		newRelSet := newRel.(*schema.Set)
+		relToDelete := toStringList(oldRelSet.Difference(newRelSet).List())
+		relToCreate := toStringList(newRelSet.Difference(oldRelSet).List())
+
+		for _, relDn := range relToDelete {
+			err = aciClient.DeleteRelationdhcpRsProvFromDHCPRelayPolicy(dhcpRelayP.DistinguishedName, relDn)
+			if err != nil {
+				return err
+			}
+
+		}
+
+		for _, relDn := range relToCreate {
+			err = aciClient.CreateRelationdhcpRsProvFromDHCPRelayPolicy(dhcpRelayP.DistinguishedName, relDn)
+			if err != nil {
+				return err
+			}
+			d.Partial(true)
+			d.SetPartial("relation_dhcp_rs_prov")
+			d.Partial(false)
+
+		}
+
+	}
+
+	d.SetId(dhcpRelayP.DistinguishedName)
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+
+	return resourceAciDHCPRelayPolicyRead(d, m)
+
+}
+
+func resourceAciDHCPRelayPolicyRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
+
+	aciClient := m.(*client.Client)
+
+	dn := d.Id()
+	dhcpRelayP, err := getRemoteDHCPRelayPolicy(aciClient, dn)
+
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+	setDHCPRelayPolicyAttributes(dhcpRelayP, d)
+
+	dhcpRsProvData, err := aciClient.ReadRelationdhcpRsProvFromDHCPRelayPolicy(dn)
```

Comment:
You have to add logic which compares two list of string and set the value accordingly